### PR TITLE
feat(jobs-api): adding mock API calls for jobs listing and status check

### PR DESCRIPTION
**Pull request by str-anger**
--

### CODE REVIEW

File: aqueductcore/backend/routers/graphql/types.py [hunk 1]
```diff
@@ -173,3 +174,34 @@ def from_extension(extension: Extension):
...
+@strawberry.enum
+class TaskStatus(Enum):
+    """Statuses of task execution"""
+
+    FAILURE = "failure"
+    PENDING = "pending"
+    RECEIVED = "received"
+    REVOKED = "revoked"
+    STARTED = "started"
+    SUCCESS = "success"
```
Comment (safeamiiir) [inline aqueductcore/backend/routers/graphql/types.py:188]: Are these Flower API states? if yes we can have them like as they are and later in the frontend they will be labeled as it's in the design.

File: aqueductcore/backend/routers/graphql/query_schema.py [hunk 2]
```diff
@@ -106,3 +113,78 @@ async def tags(
...
+            task_id=task_id,
+            eid=f"20240801-{position}",
+            author=f"Tom-{position // 10}",
+            extension_name=f"Mock extension name-{position // 5}",
+            action_name=f"Mock action name-{position % 6}",
+            started_time=datetime.now(),
+            receive_time=datetime(2023, 12, 1 + position % 30, 23, 59, position, 999),
+            task_runtime=timedelta_to_string(
+                datetime.now() - datetime(2023, 12, 1 + position % 30, 23, 59, 59, 999)
+            ),
```
Comment (safeamiiir) [inline aqueductcore/backend/routers/graphql/query_schema.py:134]: Why do we want to do this conversion in the server?
We can have this inexpensive computation in the frontend and it's easier to be converted based on the design in different places.
time in seconds should be fine.

File: aqueductcore/backend/routers/graphql/inputs.py [hunk 3]
```diff
@@ -61,3 +61,14 @@ class ExperimentRemoveInput:
     """Input type to remove experiment"""
 
     uuid: UUID
+
+
+@strawberry.input
+class TasksFilterInput:
+    """Filter for tasks list"""
+
+    author_name: Optional[str] = None
```
Comment (samiralavi) [inline aqueductcore/backend/routers/graphql/inputs.py:70]: `author` is a new keyword that we have not used so far, let's user `username` instead which we already have

File: aqueductcore/backend/routers/graphql/inputs.py [hunk 4]
```diff
@@ -61,3 +61,14 @@ class ExperimentRemoveInput:
     """Input type to remove experiment"""
 
     uuid: UUID
+
+
+@strawberry.input
+class TasksFilterInput:
```
Comment (safeamiiir) [inline aqueductcore/backend/routers/graphql/inputs.py:67]: This change does help with the experiment details page design, well done Stas!

File: aqueductcore/backend/routers/graphql/inputs.py [hunk 5]
```diff
@@ -61,3 +61,14 @@ class ExperimentRemoveInput:
...
+    """Filter for tasks list"""
+
+    author_name: Optional[str] = None
+    extension_name: Optional[str] = None
+    experiment_id: Optional[str] = None
```
Comment (samiralavi) [inline aqueductcore/backend/routers/graphql/inputs.py:72]: We have an `ExperimentIdentifierInput` type, replace `str` with it.

File: aqueductcore/backend/routers/graphql/query_schema.py [hunk 6]
```diff
@@ -106,3 +113,78 @@ async def tags(
...
+        if task.task_state in [TaskStatus.SUCCESS, TaskStatus.FAILURE]:
+            task.result_code = [TaskStatus.SUCCESS, TaskStatus.FAILURE].index(task.task_state)
+            task.stderr_text = "Some text"
+            task.stdout_text = "Some text 2"
+            task.ended_time = datetime.now()
+        return task
+
+    # pylint: disable=unused-argument
+    @strawberry.field
+    async def task_status(
```
Comment (samiralavi) [inline aqueductcore/backend/routers/graphql/query_schema.py:150]: For coherency, please follow the naming style we implemented for experiments and change it to `task`.

File: aqueductcore/backend/routers/graphql/types.py [hunk 7]
```diff
@@ -173,3 +174,34 @@ def from_extension(extension: Extension):
...
+
+
+@strawberry.type
+class TaskInfo:
+    """Full information about the scheduled task."""
+
+    task_id: UUID = strawberry.field(description="Unique identifier of the task.")
+    eid: str = strawberry.field(description="Experiment the task is associated with.")
```
Comment (safeamiiir) [inline aqueductcore/backend/routers/graphql/types.py:196]: Not sure if it's a good practice in GraphQL or not, but can we enable this query being able to ask for more info on the experiment just here?

it saves one more request in the frontend, and we have the experiment object here already.

like:
```
experiment {
  uuid
  eid
  title
  ...
}
```

File: aqueductcore/backend/routers/graphql/query_schema.py [hunk 8]
```diff
@@ -106,3 +113,78 @@ async def tags(
...
+    ) -> Optional[TaskInfo]:
+        """ Returns information about the task
+        with a given identifier. If id is unknown,
+        returns None.
+        """
+        return Query._get_mock_task(task_id)
+
+    # pylint: disable=unused-argument
+    @strawberry.field
+    async def task_runs(
```
Comment (samiralavi) [inline aqueductcore/backend/routers/graphql/query_schema.py:163]: For coherency, please follow the naming style we implemented for experiments and change it to `tasks`.

**Review by safeamiiir — CHANGES_REQUESTED** [review]

File: aqueductcore/backend/routers/graphql/types.py [hunk 9]
```diff
@@ -173,3 +174,34 @@ def from_extension(extension: Extension):
...
+    author: Optional[str] = strawberry.field(description="User, who stated the task.")
```
Comment (samiralavi) [inline aqueductcore/backend/routers/graphql/types.py:197]: `user` or `username`

**Review by samiralavi — CHANGES_REQUESTED** [review]
Good work @str-anger, I requested a few changes to the naming convention.

File: aqueductcore/backend/routers/graphql/inputs.py (hunk 4 above)
Comment (jatinriverlane) [inline aqueductcore/backend/routers/graphql/inputs.py:67]: How about having action_name as well in filter to allow users to narrow down the list more

File: aqueductcore/backend/routers/graphql/query_schema.py (hunk 2 above)
Comment (str-anger) [reply aqueductcore/backend/routers/graphql/query_schema.py:134]: As you wish =)

File: aqueductcore/backend/routers/graphql/types.py (hunk 1 above)
Comment (str-anger) [reply aqueductcore/backend/routers/graphql/types.py:188]: These are from requirements. @jatinriverlane are they Celery statuses?

File: aqueductcore/backend/routers/graphql/types.py (hunk 7 above)
Comment (safeamiiir) [reply aqueductcore/backend/routers/graphql/types.py:196]: I was reading about it and realised the [nesting](https://docs.github.com/en/graphql/guides/migrating-from-rest-to-graphql#example-nesting) is the better way with GraphQL. It's more aligned with its philosophy to "ask for what you need".

File: aqueductcore/backend/routers/graphql/types.py (hunk 1 above)
Comment (jatinriverlane) [reply aqueductcore/backend/routers/graphql/types.py:188]: Flower just sends the states as received from Celery. i.e. Yes, they are Celery statuses.

**Review by jatinriverlane — APPROVED** [review]
Thanks @str-anger

**Review by safeamiiir — APPROVED** [review]

**Review by samiralavi — APPROVED** [review]

**Pull request merged**